web/src/features/AppVersionHistory/api/getVersions.jsx

```diff
@@ -0,0 +1,190 @@
+import { useMetadata } from "@src/stores";
+import { useIsHelmManaged } from "@src/components/hooks";
+
+async function getVersions({
+  accessToken = Utilities.getToken(),
+  apiEndpoint = process.env.API_ENDPOINT,
+  _fetch = fetch,
+  currentPage = 0,
+  pageSize = 20,
+  slug,
+} = {}) {
+  try {
+    const res = await _fetch(
+      `${apiEndpoint}/app/${slug}/versions?currentPage=${currentPage}&pageSize=${pageSize}&pinLatestDeployable=true`,
+      {
+        headers: {
+          Authorization: accessToken,
+          "Content-Type": "application/json",
+        },
+        method: "GET",
+      }
+    );
+    if (!res.ok) {
+      if (res.status === 401) {
+        Utilities.logoutUser();
+        return null;
+      }
+      throw Error(`Failed to fetch apps with status ${res.status}`);
+    }
+    return await res.json();
+  } catch (err) {
+    throw Error(err);
+  }
+}
+
+// TODO: refactor this function so that the airgapped / nonairgapped are separate
+function getVersionsSelectorForKotsManaged({ versions, currentApp, metadata }) {
+  console.log("kots managed versions");
+
+  const downstream = currentApp?.downstream;
+
+  const versionHistory = versions?.versionHistory.map((version) => {
+    const isCurrentVersion =
+      version.sequence === downstream?.currentVersion?.sequence;
+    const isDeploying = version.status === "deploying";
+    const isPastVersion = find(downstream?.pastVersions, {
+      sequence: version.sequence,
+    });
+    const needsConfiguration = version.status === "pending_config";
+    const isRollback =
+      isPastVersion && version.deployedAt && currentApp?.allowRollback;
+    const isRedeploy =
+      isCurrentVersion &&
+      (version.status === "failed" || version.status === "deployed");
+    const canUpdateKots =
+      version.needsKotsUpgrade && !metadata?.isAirgap && !metadata?.isKurl;
+
+    let statusLabel = "";
+
+    if (needsConfiguration) {
+      statusLabel = "Configure";
+    } else if (downstream?.currentVersion?.sequence == undefined) {
+      if (canUpdateKots) {
+        statusLabel = "Upgrade";
+      } else {
+        statusLabel = "Deploy";
+      }
+    } else if (isRedeploy) {
+      statusLabel = "Redeploy";
+    } else if (isRollback) {
+      statusLabel = "Rollback";
+    } else if (isDeploying) {
+      statusLabel = "Deploying";
+    } else if (isCurrentVersion) {
+      statusLabel = "Deployed";
+    } else {
+      if (canUpdateKots) {
+        statusLabel = "Upgrade";
+      } else {
+        statusLabel = "Deploy";
+      }
+    }
+    return {
+      version,
+      statusLabel,
+    };
+  });
+
+  return {
+    ...versions,
+    versionHistory,
+  };
+}
+
+// TODO: refactor this function so that the airgapped / nonairgapped are separate
+function getVersionsSelectorForAirgapped({ versions, currentApp, metadata }) {
+  return getVersionsSelectorForKotsManaged({ versions, currentApp, metadata });
+}
+
+function getVersionsSelectorForHelmManaged({ versions }) {
```

Comment:
kots managed uses a lot of factors to determine if something is or isn't a rollback. for helm managed we're using the sequence. This might change later. 

Helm charts have a `version` and a `sequence`. the version is what is in the chart and the sequence is a index of the install.

So when you run helm upgrade, it will keep meta data for a few previous installations so you can use `rollback` if needed